prime-router/src/main/kotlin/fhirengine/translation/hl7/utils/ConstantResolver.kt

```diff
@@ -130,24 +129,25 @@ class FhirPathCustomResolver : FHIRPathEngine.IEvaluationContext, Logging {
         }
 
         // Evaluate the constant before it is used.
-        val value = if (constantValue.isNullOrBlank()) null
+        val values = if (constantValue.isNullOrBlank()) null
         else {
             val values = FhirPathUtils.evaluate(appContext, appContext.focusResource, appContext.bundle, constantValue)
             if (values.isEmpty()) {
                 null
-            } else if (values.size != 1) {
-                // TODO: Support multiple values with the updated function return type and remove our custom solution
-                throw SchemaException("Constant $name must resolve to one value, but had ${values.size}.")
             } else {
-                logger.trace("Evaluated FHIR Path constant $name to: ${values[0]}")
-                // Convert string constants that are whole integers to Integer type to facilitate math operations
-                if (values[0] is StringType && StringUtils.isNumeric(values[0].primitiveValue())) {
-                    IntegerType(values[0].primitiveValue())
-                } else values[0]
+                logger.trace("Evaluated FHIR Path constant $name to: $values")
+                val cleanedValues = mutableListOf<Base>()
+                for (value in values) {
+                    if (value is StringType && StringUtils.isNumeric(values[0].primitiveValue())) {
```

Comment:
You need to add unit tests to test this code with multiple values both strings that can be numeric and not numeric.

Reply:
There already exist unit tests that check the types:

`assertThat(result[0] is IntegerType).isTrue()`

`assertThat(result[0]).isInstanceOf(StringType::class.java)`

Reply:
Also check this in the new unit test I wrote.